completions.go

```diff
@@ -1,34 +1,88 @@
 package kingpin
 
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type completion struct {
+	flags []string
+	words []string
+}
+
+const (
```

Comment:
Okay, I understand what they're for now. I think a better approach would be to make dedicated `bool` fields in `completion` for the two flags: `dir` and `file`, for example.